src/main/java/com/onarandombox/MultiverseCore/utils/WorldManager.java

```diff
@@ -110,52 +121,47 @@ public boolean cloneWorld(String oldName, String newName, String generator) {
 
         // Make sure the new world doesn't exist outside of multiverse.
         if (newWorldFile.exists()) {
+            Logging.warning("File for new world '%s' already exists", newName);
             return false;
         }
 
-        unloadWorld(oldName);
+        MVWorld oldWorld = (MVWorld) this.getMVWorld(oldName);
+        Environment environment = oldWorld.getEnvironment();
+        String seedString = oldWorld.getSeed() + "";
+        WorldType worldType = oldWorld.getWorldType();
+        Boolean generateStructures = null; // TODO actually get a value for this?
```

Comment:
Yes. However, I'll probably change this to grab a value from the world if loaded. Would it be possible to cache it to add support for unloaded worlds as well?